frontend/src/hooks/mutations/useMutateParticipateIn.ts

```diff
@@ -0,0 +1,38 @@
+import useToast from "../common/useToast";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import QUERY_KEYS from "@/apis/queryKeys";
+import { postParticipateIn } from "@/apis/rooms.api";
+
+interface useMutateParticipateInResult {
+  handleParticipateIn: (roomId: number) => void;
+}
+
+const useMutateParticipateIn = (): useMutateParticipateInResult => {
+  const queryClient = useQueryClient();
+  const { openToast } = useToast();
+
+  const onSuccess = () => {
+    queryClient.invalidateQueries({
+      queryKey: [QUERY_KEYS.PARTICIPATED_ROOM_LIST, QUERY_KEYS.OPENED_ROOM_LIST],
+    });
+  };
+
+  const onError = (error: Error) => {
+    openToast(error.message);
+  };
+
+  const participateIn = useMutation({
+    mutationFn: (roomId: number) => postParticipateIn(roomId),
+    onSuccess,
+    onError: (error) => onError(error),
+    networkMode: "always",
+  });
+
+  const handleParticipateIn = (productId: number) => {
+    participateIn.mutate(productId);
+  };
```

Comment:
handleParticipatedIn 은 큰 역할이 없는거 같은데 차라리 아래와 같이 useMutation 을 그대로 반환하는건 어떤가요?
```suggestion
  return useMutation({
    mutationFn: (roomId: number) => postParticipateIn(roomId),
    onSuccess,
    onError: (error) => onError(error),
    networkMode: "always",
  });
```

Reply:
네!! 근데 추후에 post 뿐만 아니라 다른 mutation이 생길 수도 있어서 아래와 같이 변경하였습니다😊

```ts
  const postParticipateInMutation = useMutation({
    mutationFn: (roomId: number) => postParticipateIn(roomId),
    onSuccess: () =>
      handleMutateSuccess([QUERY_KEYS.PARTICIPATED_ROOM_LIST, QUERY_KEYS.OPENED_ROOM_LIST]),
    onError: (error) => handleMutateError(error),
    networkMode: "always",
  });

  return { postParticipateInMutation };
```